### ISSUE TITLE
那些损失函数

### ISSUE THREAD

**Issue by LONGMIAOMIAO** [LONGMIAOMIAO/AI#20]
[那些损失函数](http://www.sohu.com/a/209682583_99950807)

**Comment by LONGMIAOMIAO** [comment]
先说结论：误差服从高斯分布的情况下， 最小二乘法等价于极大似然估计。
[最大似然估计和最小二乘法怎么理解？](https://www.zhihu.com/question/20447622)
[线性回归时，为什么最小二乘法是最好方法？](http://www.360doc.com/content/14/0919/15/14106735_410716258.shtml)

**Comment by LONGMIAOMIAO** [comment]
[交叉熵损失函数比二乘损失函数好在哪里？](http://blog.csdn.net/u012162613/article/details/44239919)
[交叉熵损失函数比二乘损失函数好在哪里？](http://blog.csdn.net/u012494820/article/details/52797916)

**Comment by LONGMIAOMIAO** [comment]
[Softmax 函数的特点和作用是什么？](https://www.zhihu.com/question/23765351)

**Comment by LONGMIAOMIAO** [comment]
[从神经网络视角看均方误差与交叉熵作为损失函数时的共同点](https://www.cnblogs.com/dengdan890730/archive/2016/05/31/5545616.html)

**Comment by LONGMIAOMIAO** [comment]
```
−t * log( y )

−∑ t * log( y ) + ( 1 − t ) * log( 1 − y )
```
这两个都是交叉熵损失函数，但是看起来长的却有天壤之别。为什么同是交叉熵损失函数，长的却不一样呢？

因为这两个交叉熵损失函数对应不同的最后一层的输出。第一个对应的最后一层是softmax，第二个对应的最后一层是sigmoid。

[两种交叉熵损失函数的异同](http://blog.csdn.net/u012436149/article/details/69660214)

**Comment by LONGMIAOMIAO** [comment]
[最大似然损失和交叉熵损失函数的联系](http://blog.csdn.net/diligent_321/article/details/53115369)
[有关熵的几个概念 及 最大似然和交叉熵的一致性](http://blog.csdn.net/yujianmin1990/article/details/71213601)

**Comment by LONGMIAOMIAO** [comment]
为什么logistic回归的损失函数使用到的交叉熵而不是最小二乘？

[关于最大似然与交叉熵损失函数和最小二乘法的思考](http://blog.csdn.net/u010223750/article/details/52747895)

**Comment by LONGMIAOMIAO** [comment]
[Linear SVM 和 LR 有什么异同？](https://www.zhihu.com/question/26768865)
[机器学习中常见的损失函数](http://blog.csdn.net/heyongluoyao8/article/details/52462400)
[逻辑回归](http://blog.csdn.net/pakko/article/details/37878837)
[支持向量机通俗导论（理解SVM的三层境界）](http://blog.csdn.net/macyang/article/details/38782399)
[为什么将LR和SVM放在一起来进行比较？](https://www.cnblogs.com/zhizhan/p/5038747.html)

**Comment by LONGMIAOMIAO** [comment]
一个简单的两层的sigmoid输出的的梯度是一样，后来发现神经网络拟合的时候，我们用的是均方误差的loss function ，而在这个lr 的推倒的时候，我同样用了均方误差的loss，因此得到一样的结果，但是通常在lr的推倒的时候，我们是直接用 最大似然估计的，然后只有当 误差的分布满足高斯分布的时候，最大似然的结果才会和最小二乘相同（loss function 为均方误差）

lr 和 svm本质不同在于loss function的不同，不过想想这几乎对所有的单层模型都使用，lr的损失函数是 cross entropy loss， adaboost的损失函数是 expotional loss ,svm是hinge loss，常见的回归模型通常用 均方误差 loss。

[LR和SVM的不同](https://www.cnblogs.com/zhizhan/p/5038747.html)

**Comment by LONGMIAOMIAO** [comment]
[盘点机器学习中那些神奇的损失函数](http://www.sohu.com/a/209682583_99950807)

**Comment by LONGMIAOMIAO** [comment]
[神经网络中常用的误差平方和损失函数是什么](http://blog.csdn.net/qq745021926/article/details/53191904)
